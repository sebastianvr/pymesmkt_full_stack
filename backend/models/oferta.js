const { DataTypes } = require('sequelize');
const db = require('../db/connection');


const Oferta = db.define(
    'Oferta',
    {
        id: {
            type: DataTypes.STRING,
            primaryKey: true,
        },
        mensaje: {
            type: DataTypes.TEXT
        },
        precioOferta: {
            type: DataTypes.INTEGER,
        },
        // ESTADOS DE OFERTA
        // DISPONIBLE: oferta que no ha sido rechazada ni pagada.
        // FINALIZADA: oferta pagada.
        procesoDeOferta: {
            type: DataTypes.ENUM('FINALIZADA', 'DISPONIBLE', 'RECHAZADA'),
            defaultValue: 'DISPONIBLE' 
        },
        // Eliminacion, SOLO MODIFICO A FALSO, AUN EXISTE!
        estado: {
            type: DataTypes.BOOLEAN,
            defaultValue: true
        },
        createdAt: {
            type: DataTypes.DATE
        },
        updatedAt: {
            type: DataTypes.DATE
        },
    },
    {
        indexes: [{ unique: true, fields: ['id'] }]
    });

module.exports = Oferta;
