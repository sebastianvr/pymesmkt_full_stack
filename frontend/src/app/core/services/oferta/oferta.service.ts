import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, Subject } from 'rxjs';
import { environment } from '../../../../environments/environment.prod';
import { tap } from 'rxjs/operators';


@Injectable({
  providedIn: 'root'
})
export class OfertaService {

  private _refresh$ = new Subject<void>();
  private url: string = environment.baseUrl

  constructor(
    private http: HttpClient
  ) { }

  get refresh$() {
    return this._refresh$;
  }

  postOferta(oferta: any): Observable<any> {
    return this.http.post<any>(`${this.url}/api/oferta/`, oferta)
  }

  getOfertasRecibidas(idUsuario: string): Observable<any> {
    return this.http.get<any>(`${this.url}/api/oferta/received/${idUsuario}`);
  }

  getOfertaById(idOferta: string): Observable<any> {
    return this.http.get<any>(`${this.url}/api/oferta/${idOferta}`);
  }

  getOfertasRealizadas(idUsuario: string): Observable<any> {
    return this.http.get<any>(`${this.url}/api/oferta/created/${idUsuario}`)
  }

  deleteOferta(idOferta: string): Observable<any> {
    return this.http.delete<any>(`${this.url}/api/oferta/${idOferta}`)
      .pipe(
        tap(() => {
          this.refresh$.next()
        })
      );
  }
  // getPymeById(id: string): Observable<any> {
  //   return this.http.get<any>(`${this.url}/api/pyme/${id}`)
  // }

  // getAllPymes(page: number = 0, size: number = 10) {
  //   return this.http.get<any>(`${this.url}/api/publicacion/?size=${size}&page=${page}`)
  // }


  // putPyme() {
  // }

}
