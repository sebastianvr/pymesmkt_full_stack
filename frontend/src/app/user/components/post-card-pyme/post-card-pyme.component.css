.card {
    transition: box-shadow 0.3s;
    /* Transici√≥n suave para la sombra */
}

.card:hover {
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
    /* Sombras al pasar el mouse sobre la tarjeta */
}

/* Image for empty publications */
.icon-image {
    width: 30%;
    height: auto;
    max-width: 150px;
}

.empty-publications-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 20px;
}

.tableResume tbody tr td {
    border-bottom: 1px solid #c8c9ca;
}

.icon-button-style {
    filter: invert(100%);
    width: 16px;
    height: 16px;
}