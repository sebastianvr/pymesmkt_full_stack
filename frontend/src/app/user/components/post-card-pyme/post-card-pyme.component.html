<div class="container">
  <div class="row">

    <div class="col-12">
      <div *ngFor="let item of publicaciones; let i = index">

        <div class="p-0">
          <div class="card cardMain  shadow-sm rounded mb-2" style="width: 18rem;">

            <div class="card-header pt-2">
              <h6 class="dateHeader text-muted"> <strong class="">Id publicación:</strong> {{item.id}}</h6>
              <!--  [ngClass]="{'alert-primary' : item.ofertasRecibidas ==='Sé el primero en ofertar', 'alert-warning' : item.ofertasRecibidas.includes('ofertas recibidas')}" -->
              <h6 class="stateHeader float-end alert m-0 p-1" role="alert">{{item.cantidadOfertasRecibidas}}</h6>
            </div>

            <div class="card-body">
              <h5 class="card-title"><a href="#" class="card-link">{{item.titulo | uppercase}}</a></h5>
              <p class="card-text">{{item.descripcion | uppercase}}</p>

              <hr style="height: 3px;">

              <table class="tableResume table table-hover table-borderless">
                <tbody>
                  <tr>
                    <td colspan="2" class="start-*"><strong>Monto Disponible</strong></td>
                    <td class="text-end">{{item.precioTotal | currency : 'CLP'}}</td>
                  </tr>
                  <tr>
                    <td colspan="2" class="start-*"><strong>Empresa</strong></td>
                    <td class="text-end">{{item.Usuario.Pyme.nombrePyme}}</td>
                  </tr>
                  <tr>
                    <td colspan="2" class="start-*"><strong>Calificación</strong></td>
                    <!-- <td class="text-end">{{item.Usuario.Calificacions[i].promedio}}</td> -->
                  </tr>
                </tbody>
              </table>


            </div>

            <div class="card-footer containerMoreInformation">

              <button class="btn btn-primary text-center" type="button" data-bs-toggle="collapse"
                [attr.data-bs-target]="'#collapseExample' + i" aria-expanded="false" aria-controls="collapseExample">
                Más información
                <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor"
                  class="bi bi-chevron-down" viewBox="0 0 16 16">
                  <path fill-rule="evenodd"
                    d="M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z" />
                </svg>
              </button>

              <div class="cardAccordion collapse mt-2" [attr.id]="'collapseExample' + i">
                <div class="card card-body col-12">
                  <h4 class="text-center my-2">Detalles de la publicación</h4>
                  <table class="tableResume table table-hover table-borderless">
                    <tbody>
                      <tr>
                        <td colspan="2" class="start-*"><strong>Fecha creación</strong></td>
                        <td class="text-end">{{item.createdAt | date : 'dd MMMM yyyy'}}</td>
                      </tr>
                      <tr *ngIf="item.fechaInicioServicio">
                        <td colspan="2" class="start-*"><strong>Fecha de inicio servicio</strong></td>
                        <td class="text-end">{{item.fechaInicioServicio | date : 'dd MMMM yyyy'}}</td>
                      </tr>
                      <tr *ngIf="item.fechaFinServicio">
                        <td colspan="2" class="start-*"><strong>Fecha de término servicio</strong></td>
                        <td class="text-end">{{item.fechaFinServicio | date : 'dd MMMM yyyy'}}</td>
                      </tr>
                      <tr>
                        <td colspan="2" class="start-*"><strong>Garantía</strong></td>
                        <td class="text-end">{{item.garantia.toString() | i18nSelect:garantiaMapa}}</td>
                      </tr>
                      <tr *ngIf="item.aniosGarantia">
                        <td colspan="2" class="start-*"><strong>Años de garantía</strong></td>
                        <td class="text-end">{{item.aniosGarantia | number }}</td>
                      </tr>
                      <tr *ngIf="item.cantidadElementos">
                        <td colspan="2" class="start-*"><strong>Cantidad</strong></td>
                        <td class="text-end">{{item.cantidadElementos}}</td>
                      </tr>
                      <tr *ngIf="item.modelo">
                        <td colspan="2" class="start-*"><strong>Modelo</strong></td>
                        <td class="text-end">{{item.modelo}}</td>
                      </tr>
                      <tr *ngIf="item.color">
                        <td colspan="2" class="start-*"><strong>Color</strong></td>
                        <td class="text-end">{{item.color | titlecase}}</td>
                      </tr>
                    </tbody>
                  </table>

                  <div class="container p-0 text-center">
                    <div class="col-md-4 col-sm-12">
                      <h3 class="fs-6">Archivos adjuntos</h3>
                      <img class="d-block mx-auto" width="50" src="../../../../assets/icons/file-earmark-arrow-down.svg"
                        alt="ile-earmark-arrow-down.svg">
                      <button class="btn btn-secondary btn-sm my-2" type="button">Descargar</button>
                    </div>
                  </div>

                  <div *ngIf="isIdsIguales(item.UsuarioId)" class="container">
                    <div class="d-grid gap-2 d-md-flex justify-content-md-end">
                      <button class="btn btn-lg btn-success" type="button" data-bs-toggle="modal"
                        data-bs-target="#exampleModal">Ofertar
                      </button>
                    </div>

                    <!-- Modal para ofertar-->

                    <form [formGroup]="formularioOferta" (ngSubmit)="enviarOferta(item.id)">
                      <div class="modal fade" id="exampleModal" data-bs-backdrop="static" tabindex="-1"
                        aria-labelledby="exampleModalLabel" aria-hidden="true">
                        <div class="modal-dialog modal-dialog-centered">
                          <div class="modal-content">
                            <div class="modal-header">
                              <!-- <h5 class="modal-title" id="exampleModalLabel">Escribe una oferta</h5> -->
                              <button type="button" class="btn-close" data-bs-dismiss="modal"
                                aria-label="Close"></button>
                            </div>

                            <div class="modal-body">
                              <div class="container">

                                <!-- FORMULARIO PARA ENVIAR EL MANSAJE DE OFERTA -->
                                <div class="col-md-12">
                                  <h3 class="fs-5 mb-2">Escribe tu propuesta y empieza a colaborar con esta empresa.
                                  </h3>
                                  <label for="descriptionPublication" class="form-label mb-0">Mensaje</label>
                                  <textarea pInputTextarea class="form-control" formControlName="mensaje"
                                    id="descriptionPublication"></textarea>
                                  <span *ngIf="campoInvalido('mensaje')" class="form-text text-danger">
                                    El mensaje es obligatorio.
                                  </span>
                                </div>
                                <br>
                                <div class="col-md-12" style="width: 300px;">
                                  <label for="maximumPrice" class="form-label mb-0 d-block">Precio de oferta</label>
                                  <div class="p-inputgroup">
                                    <span class="p-inputgroup-addon">$</span>
                                    <p-inputNumber inputId="maximumPrice" mode="decimal" locale="es-CH" currency="CLP"
                                      formControlName="precioOferta"></p-inputNumber>
                                  </div>
                                  <span *ngIf="campoInvalido('precioOferta')" class="form-text d-block text-danger">
                                    El precio es obligatorio.
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div class="modal-footer">
                              <button type="submit" class="btn btn-primary" data-bs-dismiss="modal">Enviar
                                oferta</button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </form>

                  </div>


                </div>
              </div>
            </div>
          </div>
          <br>
        </div>

      </div>
    </div>
  </div>
  <div class="row">
    <nav aria-label="Page navigation example">
      <ul class="pagination justify-content-end">
        <li class="page-item disabled">
          <a class="page-link">Anterior</a>
        </li>
        <li class="page-item"><a class="page-link" href="#">1</a></li>
        <li class="page-item"><a class="page-link" href="#">2</a></li>
        <li class="page-item"><a class="page-link" href="#">3</a></li>
        <li class="page-item">
          <a class="page-link" href="#">Siguiente</a>
        </li>
      </ul>
    </nav>
  </div>
  <!-- <div class="row">
    <p-paginator [rows]="10" [totalRecords]="120" [rowsPerPageOptions]="[10,20,30]"></p-paginator>
    <h5>Basic</h5>
    <p-paginator [rows]="10" [totalRecords]="totalRecords" [rowsPerPageOptions]="[10,20,30]"></p-paginator>
  </div> -->
</div>