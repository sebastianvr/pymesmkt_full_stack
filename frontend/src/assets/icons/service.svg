<svg version="1.0"
    xmlns="http://www.w3.org/2000/svg" width="900.000000pt" height="900.000000pt" viewBox="0 0 900.000000 900.000000" preserveAspectRatio="xMidYMid meet">

    <g transform="translate(0.000000,900.000000) scale(0.100000,-0.100000)" fill="#000000" stroke="none">
        <path d="M4135 8624 c-42 -22 -61 -41 -79 -79 -9 -18 -38 -155 -66 -305 -28
-151 -51 -274 -53 -275 -6 -5 -236 -95 -242 -95 -5 0 -107 71 -229 159 -122
87 -231 163 -243 170 -33 17 -114 13 -150 -6 -18 -10 -128 -113 -246 -230
-202 -198 -215 -214 -227 -262 -9 -40 -9 -58 1 -88 6 -21 82 -139 167 -262 84
-124 156 -230 158 -236 3 -6 -8 -37 -24 -70 -16 -33 -37 -82 -47 -110 -10 -27
-22 -54 -26 -59 -4 -4 -136 -30 -293 -56 -251 -41 -289 -50 -319 -73 -65 -49
-67 -60 -70 -409 -3 -443 -27 -412 388 -488 154 -28 283 -54 287 -58 3 -4 20
-41 36 -82 16 -41 38 -94 47 -117 l17 -41 -170 -236 c-247 -341 -249 -311 56
-618 118 -120 231 -225 250 -234 24 -11 51 -15 83 -12 51 5 49 4 437 270 l103
70 77 -35 c42 -19 96 -41 119 -47 33 -10 43 -19 47 -39 3 -14 26 -144 51 -289
48 -287 58 -313 123 -354 37 -23 40 -23 364 -23 l326 0 39 31 c21 16 46 46 55
65 9 19 38 158 63 309 25 151 47 275 48 276 4 4 183 76 213 86 29 10 39 4 255
-149 123 -88 234 -165 246 -171 12 -7 46 -12 75 -12 43 0 64 6 99 29 59 39
435 419 448 454 18 47 13 108 -12 152 -13 22 -92 136 -175 253 l-152 213 24
47 c12 26 36 80 51 120 16 39 32 72 36 72 5 0 129 20 276 45 301 50 340 63
377 129 23 40 24 50 29 341 6 333 4 352 -54 411 -41 42 -46 43 -349 94 -135
22 -255 45 -266 50 -12 4 -24 20 -28 35 -4 14 -24 67 -45 116 l-39 90 158 222
c87 123 165 237 174 255 20 40 21 104 0 145 -23 46 -432 455 -473 473 -48 20
-91 17 -142 -9 -24 -12 -140 -91 -258 -174 l-213 -152 -57 28 c-31 16 -84 39
-118 51 -35 12 -63 24 -63 26 0 42 -94 564 -106 590 -9 20 -35 48 -57 62 l-40
27 -321 3 c-277 2 -325 0 -351 -14z m461 -348 c1 0 18 -107 38 -236 21 -129
44 -254 52 -277 21 -59 60 -88 159 -118 110 -33 224 -81 304 -128 81 -48 133
-56 192 -29 24 11 130 81 234 156 105 74 195 136 200 136 6 0 46 -36 89 -79
l78 -80 -145 -203 c-80 -112 -152 -218 -162 -237 -26 -52 -15 -105 48 -228 30
-59 72 -162 93 -228 20 -66 44 -130 52 -142 32 -50 74 -64 340 -109 141 -24
258 -44 260 -44 1 0 1 -53 0 -117 l-3 -117 -260 -43 c-154 -26 -272 -51 -289
-61 -45 -26 -63 -58 -107 -186 -22 -65 -66 -171 -99 -235 -65 -126 -73 -167
-44 -224 9 -18 79 -120 155 -228 77 -108 139 -200 139 -205 0 -5 -36 -45 -79
-88 l-79 -79 -209 148 c-114 82 -226 155 -248 163 -54 18 -84 11 -210 -53 -60
-31 -165 -73 -234 -96 -130 -41 -169 -65 -191 -119 -6 -15 -30 -139 -53 -276
l-43 -249 -117 -3 c-64 -1 -117 -1 -117 0 0 2 -20 119 -44 261 -58 336 -55
331 -234 386 -56 18 -160 61 -231 96 -136 68 -193 80 -244 53 -12 -6 -113 -73
-224 -150 -110 -76 -205 -138 -210 -138 -5 0 -44 36 -87 79 -55 55 -76 83 -70
92 4 8 68 96 141 196 73 101 140 197 149 215 30 57 22 116 -29 209 -51 94
-108 236 -127 318 -13 56 -54 112 -99 136 -15 7 -118 30 -231 50 -113 20 -227
41 -253 47 l-48 10 3 116 3 116 245 39 c154 24 261 46 288 59 48 23 81 73 102
153 20 77 86 232 126 299 28 46 38 75 42 118 5 69 9 62 -169 319 -71 103 -129
191 -129 196 0 4 36 44 79 87 l79 79 212 -153 c117 -84 228 -157 247 -162 49
-13 91 -3 176 42 95 51 199 94 295 121 84 23 134 54 156 98 8 15 35 139 60
275 25 136 48 250 50 255 4 6 226 3 232 -3z"/>
        <path d="M4314 7195 c-201 -36 -358 -116 -490 -251 -273 -275 -338 -688 -164
-1034 115 -228 312 -389 567 -462 66 -19 101 -23 233 -22 139 0 164 3 243 27
253 78 440 234 554 462 142 283 120 627 -57 896 -186 284 -556 444 -886 384z
m266 -356 c184 -42 355 -212 401 -399 21 -87 16 -213 -13 -296 -149 -428 -712
-500 -962 -123 -76 115 -106 286 -72 416 49 192 201 347 389 398 62 16 194 19
257 4z"/>
        <path d="M2336 3989 c-206 -22 -341 -59 -630 -173 -304 -120 -923 -393 -1209
-531 -108 -53 -132 -85 -132 -179 0 -83 22 -124 85 -156 61 -31 122 -26 210
16 468 226 1324 589 1490 632 127 33 283 45 504 39 222 -6 333 -22 471 -68
161 -54 221 -95 410 -283 247 -246 274 -264 492 -345 311 -114 667 -175 1083
-186 326 -9 425 -27 519 -95 59 -43 81 -93 81 -182 0 -88 -18 -129 -78 -174
-109 -84 -239 -97 -942 -91 -556 5 -602 8 -930 68 -91 16 -237 46 -325 67
-201 47 -276 49 -327 10 -95 -73 -94 -220 3 -289 52 -37 474 -131 809 -181
150 -22 176 -23 845 -23 615 0 698 2 767 17 145 33 225 71 454 222 1638 1077
1541 1015 1691 1081 134 58 238 85 332 85 93 0 132 -18 179 -80 39 -50 39 -81
1 -154 -64 -123 -151 -193 -646 -518 -1443 -947 -2368 -1535 -2588 -1645 -209
-105 -401 -114 -655 -31 -126 41 -220 87 -1460 715 -195 99 -385 189 -422 201
-200 64 -373 49 -555 -48 -63 -34 -220 -132 -521 -327 -91 -58 -121 -102 -122
-176 -1 -125 128 -212 243 -163 18 7 146 86 285 176 292 186 343 213 427 218
89 6 147 -17 550 -223 1260 -641 1366 -691 1560 -740 369 -94 652 -43 1020
183 50 30 146 89 215 130 408 246 2298 1471 2615 1696 238 168 378 341 420
517 19 80 19 126 0 214 -22 104 -59 171 -139 250 -171 170 -417 210 -703 114
l-63 -21 -75 70 c-84 78 -176 129 -300 163 -118 34 -336 34 -451 1 l-82 -24
-63 52 c-80 64 -200 124 -304 151 -156 41 -373 27 -565 -37 -137 -46 -139 -47
-952 -556 -175 -109 -327 -198 -338 -198 -54 0 -346 84 -459 131 -34 15 -87
61 -201 177 -85 86 -192 183 -236 215 -226 159 -507 257 -824 286 -131 12
-348 12 -464 0z m4009 -378 c32 -12 62 -24 66 -28 5 -5 -176 -122 -402 -262
l-409 -253 -88 17 c-48 9 -156 19 -240 22 -123 5 -151 8 -140 18 19 19 625
394 678 421 60 30 185 72 252 83 80 15 215 6 283 -18z m873 -175 c101 -36 117
-20 -256 -264 -185 -121 -463 -304 -617 -405 l-280 -184 -21 66 c-23 75 -46
122 -80 168 -13 17 -24 34 -24 36 0 3 62 43 138 90 75 46 205 126 287 178 411
255 469 289 544 313 42 14 87 26 101 28 41 5 163 -10 208 -26z"/>
    </g>
</svg>
