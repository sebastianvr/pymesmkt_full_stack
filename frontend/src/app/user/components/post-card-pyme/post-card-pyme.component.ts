import { Component, Input, OnInit } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { ModalDismissReasons, NgbModal } from '@ng-bootstrap/ng-bootstrap';
import { ModalOfferDetailComponent } from '../modal-offer-detail/modal-offer-detail.component';
import { AuthService } from 'src/app/core/services/auth/auth.service';

@Component({
  selector: 'app-post-card-pyme',
  templateUrl: './post-card-pyme.component.html',
  styleUrls: ['./post-card-pyme.component.css'],
})
export class PostCardPymeComponent implements OnInit {
  @Input() publicaciones: any;
  @Input() isLoading: any;

  offerForm!: FormGroup;
  idUser: string = this.authService.usuario.id;
  closeResult!: string;
  
  garantiaMapa = {
    'true': 'Si',
    'false': 'No',
  };

  constructor(
    private modalService: NgbModal,
    private authService: AuthService,
  ) { }

  ngOnInit(): void {
    console.log('this.publicaciones', this.publicaciones);
  }

  isIdsIguales(itemId: string): boolean {
    if (this.idUser === itemId) {
      return true;
    }
    return false;
  }

  public openOffer(idOffer: string) {
    console.log({ idOffer });
    const modalRef = this.modalService.open(ModalOfferDetailComponent, { size: 'lg' });
    modalRef.componentInstance.idOffer = idOffer;
    modalRef.componentInstance.respondedUserId = this.idUser;

    modalRef.result.then(
      (result) => {
        this.closeResult = `Closed with: ${result}`;
      },
      (reason) => {
        this.closeResult = `Dismissed ${this.getDismissReason(reason)}`;
      }
    );
  }

  private getDismissReason(reason: any): string {
    if (reason === ModalDismissReasons.ESC) {
      return 'by pressing ESC';
    } else if (reason === ModalDismissReasons.BACKDROP_CLICK) {
      return 'by clicking on a backdrop';
    } else {
      return `with: ${reason}`;
    }
  }
}
