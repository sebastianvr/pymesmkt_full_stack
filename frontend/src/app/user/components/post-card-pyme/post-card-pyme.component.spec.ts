import { ComponentFixture, TestBed } from '@angular/core/testing';

import { PostCardPymeComponent } from './post-card-pyme.component';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { OfertaService } from '../../../core/services/oferta/oferta.service';

describe('PostCardPymeComponent', () => {
  let component: PostCardPymeComponent;
  let fixture: ComponentFixture<PostCardPymeComponent>;

  let compiled : HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [PostCardPymeComponent],
      imports: [
        ReactiveFormsModule,
        RouterTestingModule,
        HttpClientTestingModule,
        NgbModule
      ],
      providers: [
        OfertaService
      ]
    })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(PostCardPymeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();

    compiled  = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('Debe hacer match con el snapshot', () => { 
    expect(compiled.innerHTML).toMatchSnapshot()
   })
});
