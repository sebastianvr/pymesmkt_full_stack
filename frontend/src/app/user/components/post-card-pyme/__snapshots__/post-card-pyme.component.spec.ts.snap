// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCardPymeComponent Debe hacer match con e snapshot 1`] = `
"<div class=\\"container\\"><div class=\\"row\\"><ngb-pagination role=\\"navigation\\" class=\\"d-flex justify-content-center\\" ng-reflect-page=\\"0\\" ng-reflect-page-size=\\"2\\"><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><ul class=\\"pagination\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Previous\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">«</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"\\"
}--><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Next\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">»</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></ul></ngb-pagination></div><div class=\\"row\\"><div class=\\"col-12\\"><!--bindings={}--></div></div><div class=\\"row\\"><ngb-pagination role=\\"navigation\\" class=\\"d-flex justify-content-center\\" ng-reflect-page=\\"0\\" ng-reflect-page-size=\\"2\\"><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><ul class=\\"pagination\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Previous\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">«</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"\\"
}--><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Next\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">»</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></ul></ngb-pagination></div><!--bindings={
  \\"ng-reflect-ng-if-else\\": \\"[object Object]\\"
}--><!--container--></div>"
`;

exports[`PostCardPymeComponent Debe hacer match con el snapshot 1`] = `
"<div class=\\"container\\"><div class=\\"row\\"><ngb-pagination role=\\"navigation\\" class=\\"d-flex justify-content-center\\" ng-reflect-page=\\"0\\" ng-reflect-page-size=\\"2\\"><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><ul class=\\"pagination\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Previous\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">«</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"\\"
}--><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Next\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">»</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></ul></ngb-pagination></div><div class=\\"row\\"><div class=\\"col-12\\"><!--bindings={}--></div></div><div class=\\"row\\"><ngb-pagination role=\\"navigation\\" class=\\"d-flex justify-content-center\\" ng-reflect-page=\\"0\\" ng-reflect-page-size=\\"2\\"><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><!--container--><ul class=\\"pagination\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Previous\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">«</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"\\"
}--><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--><li class=\\"page-item disabled\\"><a aria-label=\\"Next\\" href=\\"\\" class=\\"page-link\\" tabindex=\\"-1\\" aria-disabled=\\"true\\"><span aria-hidden=\\"true\\">»</span><!--bindings={
  \\"ng-reflect-ng-template-outlet-context\\": \\"[object Object]\\"
}--></a></li><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></ul></ngb-pagination></div><!--bindings={
  \\"ng-reflect-ng-if-else\\": \\"[object Object]\\"
}--><!--container--></div>"
`;
